openquake/engine/engine.py

```diff
@@ -244,3 +247,27 @@ def _do_run_calc(calc, exports, hazard_calculation_id, **kw):
     with calc._monitor:
         calc.run(exports=exports, hazard_calculation_id=hazard_calculation_id,
                  close=False, **kw)  # don't close the datastore too soon
+
+
+def version_triple(tag):
+    """
+    returns: a triple of integers from a version tag
+    """
+    groups = re.match(r'v?(\d+)\.(\d+)\.(\d+)', tag).groups()
+    return tuple(int(n) for n in groups)
+
+
+def check_latest_version(oqparam):
+    """
+    Check if there is a newer version of the engine
+    """
+    try:
+        json = requests.get(GITHUB + '/releases/latest', timeout=0.5).json()
+        tag_name = json['tag_name']
+        current = version_triple(__version__)
+        latest = version_triple(json['tag_name'])
+    except:  # page not available or wrong version tag
+        return
+    if current < latest:
+        logs.LOG.warn('Version %s of the engine is available, but you are '
```

Comment:
Would it be possible to just return a string in the function instead of printing `log.LOG.warn`? In this way the function may be reused in the future by an API endpoint (i.e. via QGIS).

Reply:
Done